model/core/mem_addrtype.sail

```diff
@@ -24,3 +24,9 @@ function sub_virtaddr_xlenbits(Virtaddr(addr) : virtaddr, offset : xlenbits) ->
 overload operator - = { sub_virtaddr_xlenbits }
 
 overload bits_of = { bits_of_physaddr, bits_of_virtaddr }
+
+let physaddr_bits : range(13, if xlen == 32 then 34 else 56) = {
+  let pa_bits : range(13, 56) = config memory.physaddr_bits;
+  assert(xlen == 64 | pa_bits <= 34);
```

Comment:
This assert is not good.  It trips when running `--validate-config` with an out-of-range setting (e.g. 35 for a RV32 config):
```
$ ./build/c_emulator/sail_riscv_sim --validate-config --config rv32_test.json 
Assertion failed: core/mem_addrtype.sail:30.35-30.36
```
Possibly try the constraint approach @nadime15 suggested?